web_app/order_books/haiko/main.py

```diff
@@ -0,0 +1,246 @@
+import asyncio
+from decimal import Decimal
+
+from web_app.order_books.abstractions import OrderBookBase
+from web_app.order_books.constants import TOKEN_MAPPING
+from web_app.order_books.haiko.api_connector import HaikoAPIConnector, HaikoBlastAPIConnector
+from web_app.order_books.haiko.histogram import Histogram
+from web_app.order_books.haiko.logger import get_logger
+
+
+class HaikoOrderBook(OrderBookBase):
+    DEX = "Haiko"
+
+    def __init__(self, token_a, token_b, apply_filtering: bool = False):
+        """
+        Initialize the HaikoOrderBook object.
+        :param token_a: baseToken hexadecimal address
+        :param token_b: quoteToken hexadecimal address
+        :param apply_filtering: bool - If True apply min and max price filtering to the order book data
+        """
+        super().__init__(token_a, token_b)
+        self.haiko_connector = HaikoAPIConnector()
+        self.blast_connector = HaikoBlastAPIConnector()
+        self.apply_filtering = apply_filtering
+        self.logger = get_logger("./logs", echo=True)
+
+        self.token_a_name = TOKEN_MAPPING.get(token_a).name
```

Comment:
Maybe it would be better to handle the `None` the `get` method returns with some explicit error message.